app/src/androidTest/java/com/example/campusguide/DirectionsSystemTest.kt

```diff
@@ -0,0 +1,202 @@
+        if(navigateButton.exists() && navigateButton.isEnabled) {
+            navigateButton.click()
+        }
+
+        // Wait until the ChooseDestinationOptions dialog appears
+        device.wait(Until.hasObject(By.desc("chooseDestinationOptionsLayout")), TIMEOUT)
+
+        // Click on the Search For Location button
+        val searchForLocationButton: UiObject = device.findObject(UiSelector().descriptionContains("searchForLocationButton"))
+        if(searchForLocationButton.exists() && searchForLocationButton.isEnabled) {
+            searchForLocationButton.click()
+        }
+
+        // Wait until the CustomSearchActivity activity is launched
+        device.wait(Until.hasObject(By.desc("customSearchLayout")), TIMEOUT)
+
+        // Search for Concordia University
+        // Note: For some reason, the regular setText() method doesn't work on a SearchView
+        // So I had to resort to legacySetText()
+        val searchTextField: UiObject = device.findObject(UiSelector().descriptionContains("searchView"))
+        searchTextField.legacySetText("Concordia University")
+
+        // Select the first option
+        val firstResult: UiObject = device.findObject(UiSelector().className("android.widget.TextView").text("Concordia University"))
+        if (firstResult.exists() && firstResult.isEnabled) {
+            firstResult.click()
+        }
+
+        // Wait until the ChooseOriginOptions dialog appears
+        device.wait(Until.hasObject(By.desc("chooseOriginOptionsLayout")), TIMEOUT)
+
+        // Click on the Use Current Location button
+        val useCurrentLocationButton: UiObject = device.findObject(UiSelector().descriptionContains("useCurrentLocationButton"))
+        if(useCurrentLocationButton.exists() && useCurrentLocationButton.isEnabled) {
+            useCurrentLocationButton.click()
+        }
+
+        // Wait until the map of DirectionsActivity is loaded
+        device.wait(Until.hasObject(By.desc("directions_activity_map ready")), TIMEOUT)
+
+        // Pick a random travel mode
+        val travelMode = travelModes.random()
+
+        // Click on the travel mode button
+        val travelModeButton: UiObject = device.findObject(UiSelector().descriptionContains(travelMode))
+        if (travelModeButton.exists() && travelModeButton.isEnabled) {
+            travelModeButton.click()
+        }
+
+        // Wait until route is displayed
+        val markerStart: UiObject = device.findObject(UiSelector().descriptionContains("Start"))
+        val markerEnd: UiObject = device.findObject(UiSelector().descriptionContains("Destination"))
+        device.wait(Until.hasObject(By.desc("Start")), TIMEOUT)
+
+        assertThat(markerStart.contentDescription, containsString("Start"))
+        assertThat(markerEnd.contentDescription, containsString("Destination"))
+        assertThat(travelModeButton.text, not("-"))
```

Comment:
Any way to assert that the start and end are at the correct latitude/longitude? Or is that too difficult?

Reply:
Not worth the effort imo